lib/types.js

```diff
@@ -201,11 +215,14 @@ function generateTypeConversion(ctx, name, idlType, argAttrs = [], parentName, e
       output.push(`if (${condition}) {}`);
     }
 
-    if (union.callback || union.object) {
-      output.push(`if (typeof ${name} === "function") {}`);
-    }
-
-    if (union.sequenceLike || union.dictionary || union.record || union.object || union.callbackInterface) {
+    if (
+      union.sequenceLike ||
+      union.dictionary ||
+      union.record ||
+      union.object ||
+      union.callback ||
+      union.callbackInterface
+    ) {
```

Comment:
Please follow the spec for union type conversion as closely as possible: https://heycam.github.io/webidl/#es-union